frameworks/CSharp/aspnetcore/Benchmarks/Data/EfDb.cs

```diff
@@ -86,11 +77,9 @@ public async Task<World[]> LoadMultipleUpdatesRows(int count)
         public async Task<List<Fortune>> LoadFortunesRows()
         {
             var result = await _dbContext.Fortune.ToListAsync();
-
-            // TODO: compiled queries are not supported in EF 3.0-preview7
-            // await foreach (var element in _fortunesQuery(_dbContext))
-
-            result.Add(new Fortune { Message = "Additional fortune added at request time." });
+            
+            await foreach (var element in _fortunesQuery(_dbContext))
```

Comment:
This block is wrong. I think the comment I left mislead you.

Reply:
can you elaborate

Reply:
This code is adding a new fortune for each existing one. We need only one added. So the whole foreach is unnecessary. Can you check how the code was before I removye the compiled queries?

Reply:
The solution is this:

```cs
public async Task<List<Fortune>> LoadFortunesRows()
{
  var result = await _fortunesQuery(_dbContext).ToListAsync();
  result.Add(new Fortune { Message = "Additional fortune added at request time." });
  result.Sort();
  
  return result;
}
```

Reply:
It has to be something like:

```
            var result = new List<Fortune>();
            var fortunes = _fortunesQuery(_dbContext).GetAsyncEnumerator();
            while (fortunes.Current != null)
            {
                result.Add(fortunes.Current);
                await fortunes.MoveNextAsync();
            }
            result.Add(new Fortune { Message = "Additional fortune added at request time." });
            result.Sort();
  
            return result;
```

because it can only give AsyncEnumerator.
Am I right?